common/common.go

```diff
@@ -72,6 +72,15 @@ func HexToBytes(value string) ([]byte, error) {
 	return hex.DecodeString(value)
 }
 
+func HexToBytesReverse(value string) ([]byte, error) {
+	u, err := hex.DecodeString(value)
+	length := len(u)
+	var x []byte = make([]byte, length)
+	for i, j := 0, length-1; i < j; i, j = i+1, j-1 {
```

Comment:
these code have a bug. if len(u) = 2m+1, then x[m] = 0, not equal to u[m].
and seems it is not necessary to allocate a new x variable. use u should be ok.